From 53d3500820989864a2f35985386a222f7e9ac716 Mon Sep 17 00:00:00 2001
From: Alexey Brodkin <abrodkin@synopsys.com>
Date: Fri, 28 Oct 2016 13:17:04 +0300
Subject: [PATCH] perf: fix building for ARCv1

Perf uses atomic options and so it is required to have atomics enabled
in toolchain.

In case of ARC atomics are enabled by default for ARCv2 but disabled for
ARCv1. Now we explicitly enable atomics for either ARC achitecture
version so perf could be successfully built.

Currently on attempt to build perf for ARCv1 you'll see tons of:
----------------->8-----------------
undefined reference to `__sync_add_and_fetch_4'
----------------->8-----------------

Still note if ARCv1 CPU is configured without LL/SC perf will crash on
execution once "llock" instruction is attempted to be executed.

Cc: Vineet Gupta <vgupta@synopsys.com>
Cc: Peter Zijlstra <peterz@infradead.org>
Cc: Arnaldo Carvalho de Melo <acme@kernel.org>
Signed-off-by: Alexey Brodkin <abrodkin@synopsys.com>
---
 tools/perf/Makefile.config | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/tools/perf/Makefile.config b/tools/perf/Makefile.config
index 72edf83d76b7..89a29909bb64 100644
--- a/tools/perf/Makefile.config
+++ b/tools/perf/Makefile.config
@@ -23,6 +23,11 @@ $(call detected_var,ARCH)
 
 NO_PERF_REGS := 1
 
+# Additional ARCH settings for ARC
+ifeq ($(ARCH),arc)
+  CFLAGS += -matomic
+endif
+
 # Additional ARCH settings for ppc
 ifeq ($(ARCH),powerpc)
   NO_PERF_REGS := 0
-- 
2.7.4

